src/main/java/org/jabref/gui/SendAsEMailAction.java

```diff
@@ -107,19 +105,32 @@ private String sendEmail() throws Exception {
             }
         }
 
-        String mailTo = "?Body=".concat(rawEntries.toString());
-        mailTo = mailTo.concat("&Subject=");
-        mailTo = mailTo.concat(preferencesService.getExternalApplicationsPreferences().getEmailSubject());
-        for (String path : attachments) {
-            mailTo = mailTo.concat("&Attachment=\"").concat(path);
-            mailTo = mailTo.concat("\"");
-        }
-
-        URI uriMailTo = new URI("mailto", mailTo, null);
+        URI uriMailTo = getUriMailTo(rawEntries, attachments);
 
         Desktop desktop = Desktop.getDesktop();
         desktop.mail(uriMailTo);
 
         return String.format("%s: %d", Localization.lang("Entries added to an email"), entries.size());
     }
+
+    private URI getUriMailTo(StringWriter rawEntries, List<String> attachments) throws URISyntaxException {
+        StringBuilder mailTo = new StringBuilder();
+
+        mailTo.append(getEmailAddress());
+        mailTo.append("?Body=").append(getBody(rawEntries));
+        mailTo.append("&Subject=").append(getSubject());
+
+        for (String path : attachments) {
+            mailTo.append("&Attachment=\"").append(path);
+            mailTo.append("\"");
+        }
+
+        return new URI("mailto", mailTo.toString(), null);
+    }
+
+    protected abstract String getEmailAddress();
+
+    protected abstract String getSubject();
+
+    protected abstract String getBody(StringWriter rawEntries);
```

Comment:
I think, the decision is between performance or code readability.

First suggestion - then the stringwriter is converted at the caller

```suggestion
    protected abstract String getBody(String rawEntries);
```

Second suggestion: As only one sub class needs the bibtex entries, I would pass the List of bibtex entries. The standard email action can do the conversion (you have to move the code from lines 74 to 89) to there. And the kindle action can ignore it.

Reply:
Sounds good. For the second suggestion we would need both the entries and databaseContext in `getBody` - is it preferable to get those from `stateManager` in the subclass instead of passing them in as arguments?

Reply:
The stateManager can deliver the currently active database and also the currently selected Entries.

Could be a problem though if you right-click on any other entry and you want to send the clicked one. Windows explorer does unselect in this case all the other entries and only the one right-clicked on. So this should be none of your concern in this PR. I think `stateManager.getActiveDatabase()` and `stateManager.getSelectedEntries()` should be sufficient.

Reply:
Thanks for clarifying that! I'll go ahead with this approach